mlflow/sklearn/__init__.py

```diff
@@ -958,6 +959,25 @@ def patched_fit(original, self, *args, **kwargs):
                     FLAVOR_NAME, class_def, func_name, patched_fit, manage_run=True,
                 )
 
+    def patched_fn_with_autolog_disabled(original, *args, **kwargs):
+        with temporarily_disable_autologging(FLAVOR_NAME):
+            return original(*args, **kwargs)
```

Comment:
Can we just do the following?

```suggestion
        try:
            mlflow.sklearn.autolog(disable=True)
            return original(*args, **kwargs)
        finally:
            mlflow.sklearn.autolog()
```

Reply:
I'm not sure that we want to do this, since it unpatches every scikit-learn training function.

Reply:
looks like this proposal looks better.
https://github.com/mlflow/mlflow/pull/4590#discussion_r676205807

This way `sklearn.autolog(disable=True)` and then recover, we need recover all the other params. it is more troublesome.